In [33]:
# Importing Important libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')


In [34]:
df = pd.read_csv("D:\Datascince_Project\Hr-employee Prediction\Data\hr (1).csv")
df.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.24             0.79               2                   175   
1                0.40             0.49               2                   155   
2                0.45             0.45               2                   154   
3                0.46             0.73               6                   256   
4                0.14             0.54               5                   275   

   time_spend_company  Work_accident left  promotion_last_5years   Department  \
0                   5              0    0                      0           IT   
1                   3              0    1                      0           hr   
2                   3              1    1                      0      support   
3                   4              0    0                      0  product_mng   
4                   4              1    0                      0        sales   

   salary   age  
0     low   NaN  
1  medium  27.0  
2  medium   NaN  
3    high   NaN  
4    high   NaN

In [35]:
df.tail()

satisfaction_level  last_evaluation  number_project  \
14999                0.71             0.66               5   
15000                0.75             0.84               3   
15001                0.32             0.41               3   
15002                0.62             0.58               3   
15003                0.81             0.94               3   

       average_montly_hours  time_spend_company  Work_accident left  \
14999                   187                   2              0    0   
15000                   145                   3              1    0   
15001                   138                   3              1    0   
15002                   148                   2              1    0   
15003                   225                   4              0    0   

       promotion_last_5years Department  salary   age  
14999                      0      sales  medium   NaN  
15000                      0    support     low   NaN  
15001                      0  technical  medium  29.0  
15002                      0  technical     low   NaN  
15003                      0         hr  medium   NaN

In [36]:
df.shape

(15004, 11)

In [37]:
df.columns

Index(['satisfaction_level', 'last_evaluation', 'number_project',
       'average_montly_hours', 'time_spend_company', 'Work_accident', 'left',
       'promotion_last_5years', 'Department', 'salary', 'age'],
      dtype='object')

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15004 entries, 0 to 15003
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   satisfaction_level     15004 non-null  float64
 1   last_evaluation        15004 non-null  float64
 2   number_project         15004 non-null  int64  
 3   average_montly_hours   15004 non-null  int64  
 4   time_spend_company     15004 non-null  int64  
 5   Work_accident          15004 non-null  int64  
 6   left                   15004 non-null  object 
 7   promotion_last_5years  15004 non-null  int64  
 8   Department             15004 non-null  object 
 9   salary                 14975 non-null  object 
 10  age                    3080 non-null   float64
dtypes: float64(3), int64(5), object(3)
memory usage: 1.3+ MB


There are 15004 non-null values where salary has 29 null values out of 15004 and age has 11924 null values out of 15004 non null values.



In [39]:
# Checking null values
df.isna().sum()

satisfaction_level           0
last_evaluation              0
number_project               0
average_montly_hours         0
time_spend_company           0
Work_accident                0
left                         0
promotion_last_5years        0
Department                   0
salary                      29
age                      11924
dtype: int64

In [40]:
# droping the the null values
df.dropna(inplace=True)

In [41]:
# deleting the column age (if data contain more than 70% data is missing)
del df['age']

In [42]:
df.isna().sum()

satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
Department               0
salary                   0
dtype: int64

In [43]:
# checking the duplicate values/data
df.duplicated(keep="first").sum()

157

In [44]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3074 entries, 1 to 15001
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   satisfaction_level     3074 non-null   float64
 1   last_evaluation        3074 non-null   float64
 2   number_project         3074 non-null   int64  
 3   average_montly_hours   3074 non-null   int64  
 4   time_spend_company     3074 non-null   int64  
 5   Work_accident          3074 non-null   int64  
 6   left                   3074 non-null   object 
 7   promotion_last_5years  3074 non-null   int64  
 8   Department             3074 non-null   object 
 9   salary                 3074 non-null   object 
dtypes: float64(2), int64(5), object(3)
memory usage: 264.2+ KB
